Src/Json/Shared.Json/JTokenAssertions.cs

```diff
@@ -0,0 +1,234 @@
+﻿using System.Diagnostics;
+using FluentAssertions.Common;
+using FluentAssertions.Execution;
+using FluentAssertions.Formatting;
+using FluentAssertions.Primitives;
+using Newtonsoft.Json.Linq;
+
+namespace FluentAssertions.Json
+{
+    /// <summary>
+    ///     Contains a number of methods to assert that an <see cref="JToken" /> is in the expected state.
+    /// </summary>
+    [DebuggerNonUserCode]
+    public class JTokenAssertions : ReferenceTypeAssertions<JToken, JTokenAssertions>
+    {
+        static JTokenAssertions()
+        {
+            Formatter.AddFormatter(new JTokenFormatter());
+        }
+        
+        public JTokenAssertions(JToken subject)
+        {
+            Subject = subject;
+        }
+
+        /// <summary>
+        ///     Returns the type of the subject the assertion applies on.
+        /// </summary>
+#if !PORTABLE && !CORE_CLR
+        [System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
+#endif
+        protected override string Context => nameof(JToken);
+
+        /// <summary>
+        ///     Asserts that the current <see cref="JToken" /> equals the <paramref name="expected" /> element.
+        /// </summary>
+        /// <param name="expected">The expected element</param>
+        public AndConstraint<JTokenAssertions> Be(JToken expected)
+        {
+            return Be(expected, string.Empty);
+        }
+
+        /// <summary>
+        ///     Asserts that the current <see cref="JToken" /> equals the <paramref name="expected" /> element.
+        /// </summary>
+        /// <param name="expected">The expected element</param>
+        /// <param name="because">
+        ///     A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        ///     is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="reasonArgs">
+        ///     Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public AndConstraint<JTokenAssertions> Be(JToken expected, string because, params object[] reasonArgs)
+        {
+            Execute.Assertion
+                .ForCondition(JToken.DeepEquals(Subject, expected))
```

Comment:
Now that I think of it, maybe you should rename `Be` to `Equal`. That's the name I've used for collection and dictionary assertions as well. So `Equal` means they contain the same items in the same order. `Equivalent` means that they contain the same items in any order. The only exception is that there's no `Equivalent` for dictionaries because dictionaries don't define any order.